### PR TITLE
Implement try..catch in props of type function

### PR DESCRIPTION
**Pull request by cmarin001**
In a strategic move to bolster the application's stability and user experience, this pull request introduces a comprehensive enhancement across our UI components involving the implementation of try..catch blocks within all function-type props. This update aims to elevate error handling mechanisms, ensuring that exceptions within these functions are gracefully managed and do not adversely affect the overall application performance or user interaction.

### CODE REVIEW

**Review by JanuarAsprilla — APPROVED** [review]

File: src/Select/index.tsx [hunk 1]
```diff
@@ -75,13 +91,29 @@ export const Select = (props: ISelectProps) => {
...
+    try {
+      onChange && onChange(e, name);
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(error.message);
+      } else {
+        throw new Error("An unknown error occurred");
+      }
+    }
     setDisplayList(false);
```
Comment (wfercanas) [inline src/Select/index.tsx:103]: If we leave this statement here, the list won't hide ever again if an error is thrown.
Let's first setDisplayList(false) and then use the `try..catch`.

File: src/Select/index.tsx [hunk 2]
```diff
@@ -75,13 +91,29 @@ export const Select = (props: ISelectProps) => {
...
+    try {
+      onClick && onClick(e);
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(error.message);
+      } else {
+        throw new Error("An unknown error occurred");
+      }
+    }
     setDisplayList(!displayList);
```
Comment (wfercanas) [inline src/Select/index.tsx:117]: Here we have the same issue as in `onChange`. Leave the `try..catch` to the end

**Review by wfercanas — CHANGES_REQUESTED** [review]

File: src/Select/index.tsx (hunk 1 above)
Comment (cmarin001) [reply src/Select/index.tsx]: solved

File: src/Select/index.tsx (hunk 2 above)
Comment (cmarin001) [reply src/Select/index.tsx]: solved

**Review by JanuarAsprilla — APPROVED** [review]

**Review by wfercanas — APPROVED** [review]

**Pull request merged**